### PR TITLE
🔥  Feature/store : Get All Components Together 🥇 

### LINKED ISSUE #15: [4pt] Display missions - Lists render
- Use `useSelector()` Redux Hook to select the state slices and render lists of  missions in corresponding routes. i.e.:
```javascript
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
```

- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
- Render a table with the missions' data (as per design).

### LINKED ISSUE #16: Team3 [4pt] Fetch dragons - Fetch data
Fetch data from the Dragons endpoint  (https://api.spacexdata.com/v3/dragons) when a user navigates to the Dragons section.

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- id
- name
- type
- flickr_images

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### LINKED ISSUE #17: [4pt]  Fetch rockets - Fetch data
Fetch data from the Rockets endpoint  (https://api.spacexdata.com/v3/rockets) when the application starts (as Rockets is the default view).

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- id
- name
- type
- flickr_images

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### LINKED ISSUE #18: [4pt]  Fetch missions - Fetch data
Fetch data from the Missions endpoint  (https://api.spacexdata.com/v3/missions) when a user navigates to the Missions section.

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- mission_id
- mission_name
- description

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### LINKED ISSUE #19: Team3 [1pt] Create basic structure for Dragons - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### LINKED ISSUE #20: [1pt] Create basic structure for Missions - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### LINKED ISSUE #21: [1pt] Create basic structure for Rockets - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### LINKED ISSUE #22: [1pt] Create empty My profile - Setup (group task)
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- This view should be empty - you will add content in separate tasks.

### LINKED ISSUE #23: [1pt] Create website header with empty nav - Setup (group task)
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Create **empty** navigation (this will be filled in in separate tasks).

### PR DESCRIPTION
**Pull request by roniy68**
# Highlights 🥇 

## What I Have Done: 

✅ Setup React, React-Router, React-Logger
✅ Add Redux Store
✅ Add Missions , Rockets, [ TO THE PROFILE ]
✅ Add Main Canvas 
✅ Add Live Deploy
✅ BootStrap Add

# [Live Here](https://6409ddabc88f080394d0b31f--ahr-spacex.netlify.app/) 
## Why
🔴 Check if any error with setup 
🔴 Suggest Any other ideas

### Changes details
***Please add @roniy68 in your comment so that I get notified***
## Missed anything?
- [ ] Explained the purpose of this PR.
- [ ] Self reviewed the PR.
- [ ] Added or updated test cases.
- [ ] Informed of breaking changes, testing and migrations (if applicable).
- [ ] Updated documentation (if applicable).
- [ ] Attached screenshots (if applicable).

## Followed Rules: 
- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
-  Followed [Github Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md)
-  Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
-  Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
-  Followed our list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

### CODE REVIEW

**Comment by roniy68** [comment]
# ♻️ Please Ignore The Merge Conflict 🌀 
---

## It will resolve automatically after merging to development 😄

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
-    <Router>
-      <div className="App">
+    <div className="App">
+      <BrowserRouter>
         <Navbar />
         <Routes>
-          <Route path="/" element={<Home />} />
-          <Route path="missions" element={<Missions />} />
-          <Route path="profile" element={<Profile />} />
-          <Route path="*" element={<NotFound />} />
```
Comment (Victor-chiemerie) [inline src/App.js:17]: 👍  Excellent work on how you dispatch the `fetchRockets` action on the initial loading of the page, but I would suggest that you restore the file `NotFound` to present a page to the user when an error URL is typed.

File: src/components/Missions/Missions.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { useEffect } from 'react';
+import Table from 'react-bootstrap/Table';
+import Button from 'react-bootstrap/Button';
```
Comment (Victor-chiemerie) [inline src/components/Missions/Missions.jsx:3]: ✔️ your use of this react-bootstrap is an excellent work practice, I would love to learn that from you.

**Review by Victor-chiemerie — CHANGES_REQUESTED** [review]
👋 Hi @roniy68

Great job so far!
Your project is looking fantastic 👍 

However, there's a little you need to make on this project



## Highlights 🥇 

React redux used ✔️ 
APIs connected ✔️ 
No Linters error ✔️ 
Best practices for HTML and CSS ✔️ 
Best practices for JavaScript ✔️ 
Documented in a professional way ✔️ 

# Required Changes ♻️ 

check the comments under the review

# Optional Suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**


As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/App.js (hunk 1 above)
Comment (roniy68) [reply src/App.js:17]: 👍  May be we can add the Not Found page later

**Comment by roniy68** [comment]
# 😄 Thanks for your Review

**Pull request merged**